tests/common/util.rs

```diff
@@ -1191,12 +1243,525 @@ impl UCommand {
+                status,
+                self.uchild.stdout_all(),
+                self.uchild.stderr_all()
+            ),
+            Ok(None) => {}
+            Err(error) => panic!("Assertion failed with error '{:?}'", error),
+        }
+
+        self
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn is_not_alive(&mut self) -> &mut Self {
+        match self
+            .uchild
+            .as_child()
+            .try_wait()
+        {
+            Ok(None) => panic!(
+                "Assertion failed. Expected '{}' to be not running but was alive.\nstdout: {}\nstderr: {}",
+                uucore::util_name(),
+                self.uchild.stdout_all(),
+                self.uchild.stderr_all()),
+            Ok(_) =>  {},
+            Err(error) => panic!("Assertion failed with error '{:?}'", error),
+        }
+
+        self
+    }
+}
+
+pub struct UChild {
+    raw: Child,
+    bin_path: String,
+    util_name: Option<String>,
+    tmpd: Option<Rc<TempDir>>,
+    captured_stdout: Option<CapturedOutput>,
+    captured_stderr: Option<CapturedOutput>,
+    ignore_stdin_write_error: bool,
+}
+
+// TODO: rename `pipe_in` to `pipe_in_bytes(Vec<u8>)` and use convenience function `pipe_in(&str)`
+// TODO: Add method `pipe_in_fixture(&str)`: like `pipe_in` but with a fixture
+impl UChild {
+    fn new(
+        child: Child,
+        bin_path: String,
+        util_name: Option<String>,
+        tmpd: Option<Rc<TempDir>>,
+        captured_stdout: Option<CapturedOutput>,
+        captured_stderr: Option<CapturedOutput>,
+        ignore_stdin_write_error: bool,
+    ) -> Self {
+        Self {
+            raw: child,
+            bin_path,
+            util_name,
+            tmpd,
+            captured_stdout,
+            captured_stderr,
+            ignore_stdin_write_error,
+        }
+    }
+
+    /// Returns a mutable reference of the wrapped child process.
+    pub fn as_child(&mut self) -> &mut Child {
+        &mut self.raw
+    }
+
+    pub fn take_child_stdin(&mut self) -> Option<ChildStdin> {
+        self.raw.stdin.take()
+    }
+
+    pub fn delay(&mut self, millis: u64) -> &mut Self {
+        sleep(Duration::from_millis(millis));
+        self
+    }
+
+    pub fn id(&self) -> u32 {
+        self.raw.id()
+    }
+
+    /// Terminates the child process unconditionally and waits for the termination. Ignores any
+    /// errors happening during [`Child::kill`].
+    ///
+    /// # Panics
+    /// If the child process could not be terminated within 60 seconds or an error happened during
+    /// [`Child::wait_with_timeout`]
+    pub fn kill(&mut self) -> &mut Self {
+        let _ = self.raw.kill();
+        for _ in 0..60 {
+            if !self.try_alive() {
+                return self;
+            }
+            sleep(Duration::from_secs(1));
+        }
+        panic!("Killing the child process within 60 seconds failed.")
+    }
+
+    // TODO: Accept AssertionMode as parameter
```

Comment:
It's not that easy, although I think we match the common use cases the way it is right now.

```rust
ts.ucmd().run() -> CmdResult // where all output is in CmdResult

let child = ts.ucmd().run_no_wait();
// do something and/or child outputs something
child.make_assertion().with_current_output() // assert on the output from beginning to now
// do something and child outputs something
// if child is supposed to exit we can wait
child.wait() // (*) -> CmdResult and do assertions on the output from last call to make_assertion until the exit
// if child doesn't exit but we are finished with it
child.kill() // -> UChildAssertion with which we can use all, current or exact output
```

I'm not sure if it's ever used so it was still a TODO, but I wanted to improve on the `(*)` case, so you're able to use all output from start to end of the child process or the exact output and not only the current.

Reply:
Oh I see. I think that's a design question we need to answer once we find a need for it. In general, with `wait()` and `kill()` I'm expecting to get everything I can get and it would be an extremely rare situation where I might need to do something more complex. In which case I case also use the raw `Child` still.

Reply:
I'll try something, here.

Reply:
I think I found a simple solution for the `(*)` case. It would introduce 2 public methods in `UChild`. Interested or keep it the way it is for now?

Reply:
In an afterthought, I don't think, there's much gain from a usage perspective. Closing this?

Reply:
Yeah resolving this is fine